src/client.coffee

```diff
@@ -1,163 +1,158 @@
 net = require('net')
-Swank = require './lisp/swank-protocol'
 fs = require 'fs-extra'
 path = require 'path'
 log = require('loglevel').getLogger('ensime.client')
 temp = require 'temp'
+WebSocket = require("ws")
 
+temp.track()
 tempDir = temp.mkdirSync()
 getTempDir = -> tempDir
 
-module.exports =
-class Client
-  constructor: (port, @httpPort, generalMsgHandler, @serverPid = undefined) ->
-    @ensimeMessageCounter = 1
-    @callbackMap = {}
-
-    @parser = new Swank.SwankParser( (env) =>
-      log.trace("incoming: #{env}")
+module.exports = createClient = (port, httpPort, generalMsgHandler, serverPid = undefined) ->
```

Comment:
`port` can be removed, maybe? Here and where it is read in _ensime-client-startup.coffee_ ?